src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -60,6 +60,9 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
     // Promise to resolve when we have loaded our controllers
     private controllersPromise?: Promise<VSCodeNotebookController[]>;
 
+    // Listing of the controllers that we have registered
+    private registeredControllers: VSCodeNotebookController[] = [];
```

Comment:
I don't think this is necessary. You only use it one function. And in that function you have to wait on the controllers promise anyway. You could just use the result of getNotebookControllers instead.